### DISCUSSION TITLE
Precision of numbers

### DISCUSSION THREAD

**Discussion by anakrish** [open-policy-agent/community#354] [OPA and Rego]
Apologies if this is already captured somewhere. I could not find the answer to this in the docs.

What is the precision of `number` data type?
Based on experiments, a `numbers` seems to have ~64~ 63 bits of precision.

```rego
check(n) = ((n+1) - 1) == n

a = check(9223372036854775808) #2^63 => true
b = check(18446744073709551616) # 2^64 => false
```

**Comment by charlieegan3** [comment]
Hi @anakrish, OPA is using a 64bit **signed** integer in these cases. So the maximum value is `2^64-1` rather than `2^64`. Hope that helps!
